packages/vhd-lib/src/Vhd/VhdDirectory.js

```diff
@@ -167,11 +167,13 @@ export class VhdDirectory extends VhdAbstract {
     return this._writeChunk('bat', this.#blockTable)
   }
 
-  // only works if data are in the same bucket
+  // only works if data are in the same handler
   // and if the full block is modified in child ( which is the case whit xcp)
 
-  coalesceBlock(child, blockId) {
-    this._handler.copy(child.getChunkPath(blockId), this.getChunkPath(blockId))
+  async coalesceBlock(child, blockId) {
+    assert.strictEqual(this._handler, child._handler)
+    this._handler.copy(child._getBlockPath(blockId), this.getChunkPath(blockId))
+    return await this._handler.getSize(this.getChunkPath(blockId))
```

Comment:
block should always have the same size, we can return a computed size if we're confident on the copy

Reply:
This was used to indicate the quantity of data that was copied by the merge process, not a theoretical size, or we could have simply indicated `nBlocks * blockSize`.

Also, specially in the case of `VhdDirectory`, it will be different with the compression.

I'm not sure what to do here…